src/AnnenErfaring/Skjema.elm

```diff
@@ -0,0 +1,344 @@
+    if not skjema.nåværende && skjema.visFeilmeldingTilÅr then
+        Dato.feilmeldingÅr skjema.tilÅr
+
+    else
+        Nothing
+
+
+visFeilmeldingFraÅr : AnnenErfaringSkjema -> AnnenErfaringSkjema
+visFeilmeldingFraÅr (UvalidertSkjema skjema) =
+    UvalidertSkjema { skjema | visFeilmeldingFraÅr = True }
+
+
+visFeilmeldingTilÅr : AnnenErfaringSkjema -> AnnenErfaringSkjema
+visFeilmeldingTilÅr (UvalidertSkjema skjema) =
+    UvalidertSkjema { skjema | visFeilmeldingTilÅr = True }
+
+
+visFeilmeldingRolle : AnnenErfaringSkjema -> AnnenErfaringSkjema
+visFeilmeldingRolle (UvalidertSkjema skjema) =
+    UvalidertSkjema { skjema | visFeilmeldingRolle = True }
+
+
+visAlleFeilmeldinger : AnnenErfaringSkjema -> AnnenErfaringSkjema
+visAlleFeilmeldinger skjema =
+    skjema
+        |> visFeilmeldingFraÅr
+        |> visFeilmeldingTilÅr
+        |> visFeilmeldingRolle
+
+
+
+--- VALIDERING ---
+
+
+valider : AnnenErfaringSkjema -> Maybe ValidertAnnenErfaringSkjema
+valider (UvalidertSkjema uvalidert) =
+    if feilmeldingRolle uvalidert.rolle /= Nothing then
+        Nothing
+
+    else if feilmeldingBeskrivelse uvalidert.beskrivelse /= Nothing then
+        Nothing
+
+    else
+        Maybe.map2
+            (\fraÅr_ tilDato_ ->
+                ValidertSkjema
+                    { rolle = uvalidert.rolle
+                    , beskrivelse = uvalidert.beskrivelse
+                    , fraMåned = uvalidert.fraMåned
+                    , fraÅr = fraÅr_
+                    , tilDato = tilDato_
+                    , id = uvalidert.id
+                    }
+            )
+            (Dato.stringTilÅr uvalidert.fraÅr)
+            (validerTilDato uvalidert.nåværende uvalidert.tilMåned uvalidert.tilÅr)
+
+
+validerTilDato : Bool -> Måned -> String -> Maybe TilDato
+validerTilDato nåværende_ måned år =
+    if nåværende_ then
+        Just Nåværende
+
+    else
+        år
+            |> Dato.stringTilÅr
+            |> Maybe.map (Avsluttet måned)
+
+
+tilDatoMåned : TilDato -> Måned
+tilDatoMåned tilDato =
+    case tilDato of
+        Nåværende ->
+            Januar
+
+        Avsluttet måned _ ->
+            måned
+
+
+tilDatoÅr : TilDato -> String
+tilDatoÅr tilDato =
+    case tilDato of
+        Nåværende ->
+            ""
+
+        Avsluttet _ år ->
+            Dato.årTilString år
+
+
+tilUvalidertSkjema : ValidertAnnenErfaringSkjema -> AnnenErfaringSkjema
+tilUvalidertSkjema (ValidertSkjema validert) =
+    UvalidertSkjema
+        { rolle = validert.rolle
+        , beskrivelse = validert.beskrivelse
+        , fraMåned = validert.fraMåned
+        , fraÅr = Dato.årTilString validert.fraÅr
+        , nåværende = validert.tilDato == Nåværende
+        , tilMåned = tilDatoMåned validert.tilDato
+        , tilÅr = tilDatoÅr validert.tilDato
+        , visFeilmeldingFraÅr = False
+        , visFeilmeldingTilÅr = False
+        , visFeilmeldingRolle = False
+        , id = validert.id
+        }
+
+
+
+-- ENCODE --
```

Comment:
Verdens minste ting, men her har du bare to bindestreker. Overalt ellers er det 3. Jeg foretrekker 3 😄